docs: tweak explanation of median for even cardinality inputs

NickCrews opened the pull request:
Instead of qualitative/quantitative, I think we should be more direct and list the actual types.

Also, just a little more word-smithing to be explicit that we are only referring to the two middle values. This change I am less convinced is a good idea, but I still think it's a positive.

Related to https://github.com/duckdb/duckdb/issues/13655

Mytherin commented:
Thanks for the PR!

However - this description is not quite correct. The split is not `VARCHAR` and not-VARCHAR, since DuckDB supports computing the `median` over **any** type, including nested types. 

For example:

```sql
D select median({'my_nested_type': i}) from range(0,10) t(i);
┌───────────────────────────────────────────────┐
│ median(main.struct_pack(my_nested_type := i)) │
│         struct(my_nested_type bigint)         │
├───────────────────────────────────────────────┤
│ {'my_nested_type': 4}                         │
└───────────────────────────────────────────────┘
```

Types that interpolate are numeric types and date/time types. Other types (`VARCHAR`, nested types, `JSON`, `INTERVAL`) do not interpolate and select one of the two elements instead.

NickCrews commented:
ahh, yes, that makes sense. Can I phrase it just as you did? (numeric, date, and time vs everything else). Maybe it's just me that is getting confused by the quantitative/qualitative, but I'd rather be verbose and explicit over succinct but possibly confusing. But this is just a value judgement, so I'm happy to keep it as is if you like it better.

NickCrews commented:
@Mytherin I made some adjustments based on your feedback, this is ready for another review when you get the chance. Thanks!

Mytherin commented:
Thanks - the changeset is fine by me but the CI is failing. Can you run the formatter?

NickCrews committed:
docs: tweak explanation of median for even cardinality inputs

Instead of qualitative/quantitative, I think we should be more direct and say the actual types

Also, just a little more word-smithing to be explicit that we are only referring to the two middle values. This change I am less convinced is a good idea, but I still think it's a positive.

Related to https://github.com/duckdb/duckdb/issues/13655

NickCrews commented:
@Mytherin I ran the formatter, now CI is happy, I think this is good to merge if you are still happy with it. Thanks!

Mytherin commented:
Thanks!
